### ISSUE TITLE
Is there a toolkit to evaluate ADD(-S) of pose estimation results?

### ISSUE THREAD

**Issue by TouchDeeper** [YoungXIAO13/ObjectPoseEstimationDatasets#3]
Hello,

The BOP toolkit evaluate the average recall. I find most of the paper in CVPR2020 used ADD(-S) as the evaluation metric. Is there a toolkit to evaluate ADD(-S) of pose estimation results? Thanks a lot!

**Comment by YoungXIAO13** [comment]
Hello,

Thanks for your interests, I believe you can find most of the popular object pose evaluation in the code of HybridPose:
[ADD-S](https://github.com/chensong1995/HybridPose/blob/106c86cddaa52765eb82f17bd00fdc72b98a02ca/lib/utils.py#L51) and [evaluate.py](https://github.com/chensong1995/HybridPose/blob/master/src/evaluate.py)

**Comment by TouchDeeper** [comment]
Paper of BOP toolkit said the VSD is more reasonable than ADD(-S), but most of the paper of CVPR2020 use the ADD(-S), only the EPOS use VSD metric. I'm confused in choosing the evaluation metric. Could you please provide some advice? Thanks a lot!

**Comment by YoungXIAO13** [comment]
I would say ADD(-S) is a stronger evaluation metric that measures directly the distance between the actual object and the object projected by the estimated pose. This is very useful for robotic manipulation or VR tasks since we need to interact with the objects.
Meanwhile, VSD is more adequate in computer vision tasks because sometimes only a small part of the objects can be seen in the image, thus it makes sense that we only evaluate on the visible parts.

In short, these two metrics both worth testing during the evaluation, and people usually tend to test ADD(-S) for dataset like LineMod and test VSD for dataset like T-LESS.

**Comment by TouchDeeper** [comment]
Thanks for your explanation. I'm agreed with you. ADD(-S) for datasets without occluded, VSD for datasets with occluded may be more reasonable.

**Comment by wangg12** [comment]
BOP toolkit can be used to evaluate ADD(-S). https://github.com/thodan/bop_toolkit/blob/master/scripts/eval_calc_scores.py#L48

**Comment by YoungXIAO13** [comment]
> BOP toolkit can be used to evaluate ADD(-S). https://github.com/thodan/bop_toolkit/blob/master/scripts/eval_calc_scores.py#L48

Thanks for your remind!
@TouchDeeper  you can fin the ADD-S here:
https://github.com/thodan/bop_toolkit/blob/120181c015324f64b08ed1b83493c40174d00b7a/bop_toolkit_lib/pose_error.py#L164

**Comment by TouchDeeper** [comment]
@wangg12 @YoungXIAO13 , Thank you for the remind, it will save me a lot of time.
